plugin/web/extensions/geolocation/core/hooks.ts

```diff
@@ -0,0 +1,37 @@
+import { useCallback } from "react";
+
+import { CurrentLocation } from "./types";
+import { postMsg } from "./utils";
+
+const initialLocation: CurrentLocation = {
+  latitude: 5.70249,
+  longitude: 39.7622,
+  altitude: 5000,
+};
+
+export default () => {
+  let currentLocation: CurrentLocation;
+
+  const handleFlyToCurrentLocation = useCallback(() => {
+    if (navigator.geolocation) {
+      navigator.geolocation.watchPosition(
```

Comment:
Wasn't `navigator.geolocation.getCurrentPosition` more appropriate for this? `watchPosition` I think will set a listener and when the user's location changes this would fire again, which I dont think we want for this widget. We just want to set the current location on click only, right?